### PR TITLE
filter distributors before listing on checkout options

### PR DESCRIPTION
**Pull request by abdellani**
#### What? Why?

- will fix the second part of the issue on (step 2.b `4. Checkout Options`)  #10173

#### What should we test?

- Described on https://github.com/openfoodfoundation/openfoodnetwork/issues/10173
- 

#### Release notes

Changelog Category:  Technical changes

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: spec/controllers/admin/order_cycles_controller_spec.rb
Comment (Matt-Yorkley) [inline spec/controllers/admin/order_cycles_controller_spec.rb:140]: In this test the user manages the coordinating enterprise, right? It feels like the number of enterprises queried here shouldn't have changed in that case? :thinking:

**Comment by Matt-Yorkley** [comment]
I think we should add a couple more tests for the two cases mentioned in the issue that explicitly set up this scenario where there are multiple suppliers in the OC and the current user manages a supplier that's assigned to the OC but isn't the coordinator.

**Comment by Matt-Yorkley** [comment]
It's difficult to review this because it's really hard to figure which bits of code are actually loading things and from where. This is one of the reasons I dislike Angular!

I think I would prefer if we don't directly touch anything in the `OrderCyclePermissions` class and instead just change the controller which is loading the list of enterprises for this page, which as far as I can tell is here:

https://github.com/openfoodfoundation/openfoodnetwork/blob/5ec872c6fdcdd54abb05351ee92e4fb8af81089a/app/controllers/admin/enterprises_controller.rb#L121-L128

and the `@collection` it's using is here:

https://github.com/openfoodfoundation/openfoodnetwork/blob/5ec872c6fdcdd54abb05351ee92e4fb8af81089a/app/controllers/admin/enterprises_controller.rb#L178-L193

File: spec/controllers/admin/order_cycles_controller_spec.rb
Comment (abdellani) [reply spec/controllers/admin/order_cycles_controller_spec.rb:140]: I think this is coming from the queries that I added to `managed_participating_hubs` & `managed_participating_producers`

`OrderCyclesController` -> `OrderCycleSerializer` -> `OrderCyclePermissions`

Comment (abdellani) [reply spec/controllers/admin/order_cycles_controller_spec.rb:140]: I'll revert this commit since this PR will not solve that part of the issue.

File: app/helpers/order_cycles_helper.rb
Comment (mkllnk) [inline app/helpers/order_cycles_helper.rb:45]: I would hope that `Enterprise.managed_by(spree_current_user)` is cached but it wouldn't hurt to store it in a variable first. It could also be represented as a union:
```suggestion
    order_cycle.distributors & Enterprise.managed_by(spree_current_user)
```

But super-admin can see all enterprises which would make this list very long. So we could add a condition to shortcut super admin. 

Oh, hang on, I'm just realising that we completely missed the greatest feature of ActiveRecord scopes: chaining!

```suggestion
    order_cycle.distributors.managed_by(spree_current_user)
```

I reckon that this change is worth doing before merging.

**Review by mkllnk — CHANGES_REQUESTED** [review]

File: app/helpers/order_cycles_helper.rb
Comment (mkllnk) [reply app/helpers/order_cycles_helper.rb:45]: Okay, tried that and it turns out that `order_cycle.distributors` is an array, not a scope. Oh, hang on, just in the spec...

**Review by mkllnk — APPROVED** [review]
I added a few commits. That's my suggestion but I may have overlooked something. This needs another review by someone else, including @abdellani. If you are happy with this, then move it to Test Ready.

File: app/helpers/order_cycles_helper.rb
Comment (rioug) [inline app/helpers/order_cycles_helper.rb:41]: I am just trying to understand this, so spelling it out "loud" : 
So if the current user can manage the order cycle's coordinator, we show all the order cycle's distributors, I believe this covers the super admin ? 
If not we just show all the distributors that the current user can manage.

**Review by rioug — APPROVED** [review]
I believe it's good, but I must admit I don't know much about coordinator and distributor.

File: app/helpers/order_cycles_helper.rb
Comment (abdellani) [reply app/helpers/order_cycles_helper.rb:41]: Yes @rioug,  that's correct.

**Comment by dacook** [comment]
Hi @abdellani , it looks like we just need to you confirm Maikel's latest changes and move to Test Ready if you're happy with them.

**Comment by abdellani** [comment]
Ok, it looks good to me :) 
thank you @mkllnk for the updates
thanks to everyone.

**Comment by filipefurtad0** [comment]
Hey @abdellani ,

I've checked now, that supplier enterprises which do not coordinated the OC cannot view/edit payment/shipping methods from the other enterprises:

image


Also, I've checked that a distributor which is not the coordinator of the OC, can only see/edit their own shipping/payment methods:

image

This looks great!!
Thanks, merging :rocket:

**Pull request merged**
